src/autoscaler/metricscollector/collector/apppoller.go

```diff
@@ -72,10 +74,21 @@ func (ap *appPoller) startPollMetrics() {
 func (ap *appPoller) pollMetric() {
 	ap.logger.Debug("poll-metric", lager.Data{"appid": ap.appId})
 
-	containerEnvelopes, err := ap.noaaConsumer.ContainerEnvelopes(ap.appId, "bearer"+" "+ap.cfc.GetTokens().AccessToken)
-	if err != nil {
-		ap.logger.Error("poll-metric-from-noaa", err)
-		return
+	var containerEnvelopes []*events.Envelope
+	var err error
+
+	for attempt := 1; attempt <= 3; attempt++ {
+		if attempt != 0 {
```

Comment:
Remove the guard, I would want to see it for all attempts. Just remove the debug statement above.